web/src/api/stores/file.store.ts

```diff
@@ -41,34 +44,43 @@ export const useFileStore = defineStore("files", () => {
       file.astAndMappingLoaded = true;
       file.amountOfKgrams = file.amountOfKgrams || file.ast.length;
 
-      if (anonymize) {
-        const split = row.path!.split(".");
-        const extension = split[split.length - 1];
-        const name = randomNameGenerator();
-        row.path = `${name}/exercise.${extension}`;
-        extra.fullName = name;
-
-        const label = labelMap.get(extra.labels) || labelCounter;
-        if (!labelMap.has(extra.labels)) {
-          labelCounter += 1;
-        }
-        labelMap.set(extra.labels, label);
-        extra.labels = label;
-
-        if (extra.timestamp) {
-          extra.timestamp = new Date(extra.timestamp.getTime() + timeOffset);
-        }
+      // Add the label to the labels list, if it doesn't exist yet.
+      if (!labels.includes(row.label)) {
+        labels.push(extra.labels);
       }
 
+      // Store pseudo details.
+      const pseudoName = randomNameGenerator();
+      file.pseudo = {
+        path: `/exercise/${pseudoName}.${filePathExtension}`,
+        shortPath: "",
+        fullName: pseudoName,
+        timestamp: extra.timestamp ? new Date(extra.timestamp.getTime() + timeOffset) : undefined,
+        labels: String(labels.indexOf(extra.labels)) ?? "",
+      };
+
+      // Store original details.
+      file.original = {
+        path: row.path,
+        shortPath: "",
+        fullName: extra.fullName,
+        timestamp: extra.timestamp,
+        labels: extra.labels,
+      };
+
       return [row.id, row];
     });
     const files: File[] = Object.fromEntries(filesMap);
 
     // Find the common path in the files.
-    const commonPath = commonFilenamePrefix(Object.values(files));
+    const commonPath = commonFilenamePrefix(Object.values(files), (f) => f.path);

```

Comment:
This function should really only be executed for the original files, since we determine the prefix of the `pseudo` object ourselves.

I also don't think the `pseudo`-objects need a prefix, because it does not have any meaning. (See the other comment about the full path).